### ISSUE TITLE
Timeout for request/reply message

### ISSUE THREAD

**Issue by ElenaShlykova** [Particular/NServiceBus#5745]
I send a message like this:
`BusControl.Request<MessageResponse>(message, Cts.Token).ConfigureAwait(false);`
But I never receive a response if some errors occur in the consumer or if the endpoint is offline.
How to set timeout for the request/reply message? I want to receive a response after 30 seconds.

**Comment by adamralph** [comment]
@ElenaShlykova I'm not familiar with `BusControl`. It doesn't seem to be part of the NServiceBus API.

Could you be just missing an `await`?

```c#
await BusControl.Request<MessageResponse>(message, Cts.Token).ConfigureAwait(false);
```

BTW - it's usually best to raise questions via one of our [support channels](https://particular.net/support).

**Comment by andreasohlund** [comment]
@ElenaShlykova did you get this sorted out?

https://docs.particular.net/nservicebus/messaging/callbacks#cancellation is what you are looking for. So in your case pass in a cancellation token that you set to expire after 30 seconds

**Comment by adamralph** [comment]
Closing this now. @ElenaShlykova if you have further questions, it may be best to contact us via one of our support channels listed at — https://particular.net/support

Note that there are various options for both free and paid support.

I'm not entirely sure if this is directly related, but note that cancellation token support has been added in NServiceBus 8 (not yet released) — #5865.
